ingest-app/src/cmr/ingest/services/jobs.clj

```diff
@@ -218,48 +222,64 @@
+     ))}]}))
+
+(defn- add-updated-since
+ "Pull out the start and end times from a time-constraint value and associate them to a map"
+ [raw time-constraint]
+ (let [parts (clojure.string/split time-constraint, #",")
+       start-time (first parts)
+       end-time (last parts)]
+  (assoc raw :start-time start-time :end-time end-time)))
 
 (defn- process-subscriptions
   "Process each subscription in subscriptions."
   [context subscriptions time-constraint]
-  (doseq [subscription subscriptions
-         :let [email-address (get-in subscription [:extra-fields :email-address])
+  (doseq [raw_subscription subscriptions
+         :let [subscription (add-updated-since raw_subscription time-constraint)
+               email-address (get-in subscription [:extra-fields :email-address])
                sub-name (get-in subscription [:extra-fields :subscription-name])
                coll-id (get-in subscription [:extra-fields :collection-concept-id])
                query-string (-> (:metadata subscription)
                                 (json/decode true)
                                 :Query)
                query-params (create-query-params query-string)
                params1 (merge {:created-at time-constraint
-                               :collection-concept-id coll-id}
+                               :collection-concept-id coll-id
+                               :token (cmr.transmit.config/echo-system-token)}
                               query-params)
                params2 (merge {:revision-date time-constraint
-                               :collection-concept-id coll-id}
+                               :collection-concept-id coll-id
+                               :token (cmr.transmit.config/echo-system-token)}
                               query-params)]]
-      (info "Processing subscription: " sub-name)
+      (debug "Processing subscription: " sub-name " with\n" (str subscription) ".")
       (try
         (let [gran-ref1 (search/find-granule-references context params1)
               gran-ref2 (search/find-granule-references context params2)
               gran-ref (distinct (concat gran-ref1 gran-ref2))
               gran-ref-location (map :location gran-ref)]
+          (debug "gran-ref-locations for " sub-name ": " gran-ref-location)
           (when (seq gran-ref)
-            (info "Start sending email for subscription: " sub-name)
-            (postal-core/send-message {:host (email-server-host) :port (email-server-port)}
-                                      (create-email-content (mail-sender) email-address gran-ref-location subscription))
-            (info "Finished sending email for subscription: " sub-name)))
+           (let [email-content (create-email-content (mail-sender) email-address gran-ref-location subscription)
+                email-settings {:host (email-server-host) :port (email-server-port)}]
+            (postal-core/send-message email-settings email-content))))
```

Comment:
This is the third `let` block in this function, which is excessive. More than a single `let` in a function is a code smell that means it should be broken into pieces.

Reply:
I have a future ticket where I can do this, absolutely, the code needs to be simpler. Will add as a todo for ticket CMR-6632